contentcuration/search/viewsets/contentnode.py

```diff
@@ -54,168 +50,127 @@ def filter_channel_list(self, queryset, name, value):
+        return queryset.exclude(contentnode__kind_id=content_kinds.TOPIC)
 
     def filter_assessments(self, queryset, name, value):
-        return queryset.filter(kind_id=content_kinds.EXERCISE)
+        return queryset.filter(contentnode__kind_id=content_kinds.EXERCISE)
 
     def filter_created_after(self, queryset, name, value):
         date = re.search(r"(\d{4})-0?(\d+)-(\d+)", value)
         return queryset.filter(
-            created__year__gte=date.group(1),
-            created__month__gte=date.group(2),
-            created__day__gte=date.group(3),
-        )
-
-    class Meta:
-        model = ContentNode
-        fields = (
-            "keywords",
-            "languages",
-            "licenses",
-            "kinds",
-            "coach",
-            "author",
-            "resources",
-            "assessments",
+            contentnode__created__year__gte=date.group(1),
+            contentnode__created__month__gte=date.group(2),
+            contentnode__created__day__gte=date.group(3),
         )
 
 
-class SearchContentNodeViewSet(ContentNodeViewSet):
+class SearchContentNodeViewSet(ReadOnlyValuesViewset):
     filterset_class = ContentNodeFilter
     pagination_class = ListPagination
+    permission_classes = [IsAuthenticated]
+
+    field_map = {
+        "id": "contentnode__id",
+        "content_id": "contentnode__content_id",
+        "node_id": "contentnode__node_id",
+        "title": "contentnode__title",
+        "description": "contentnode__description",
+        "author": "contentnode__author",
+        "provider": "contentnode__provider",
+        "kind__kind": "contentnode__kind__kind",
+        "thumbnail_encoding": "contentnode__thumbnail_encoding",
+        "published": "contentnode__published",
+        "modified": "contentnode__modified",
+        "parent_id": "contentnode__parent_id",
+        "changed": "contentnode__changed",
+    }
+
     values = (
-        "id",
-        "content_id",
-        "node_id",
-        "title",
-        "description",
-        "author",
-        "provider",
-        "kind__kind",
+        "contentnode__id",
+        "contentnode__content_id",
+        "contentnode__node_id",
+        "contentnode__title",
+        "contentnode__description",
+        "contentnode__author",
+        "contentnode__provider",
+        "contentnode__kind__kind",
+        "contentnode__thumbnail_encoding",
+        "contentnode__published",
+        "contentnode__modified",
+        "contentnode__parent_id",
+        "contentnode__changed",
         "channel_id",
         "resource_count",
         "thumbnail_checksum",
         "thumbnail_extension",
-        "thumbnail_encoding",
-        "published",
-        "modified",
-        "parent_id",
-        "changed",
-        "location_ids",
         "content_tags",
         "original_channel_name",
     )
 
+    def get_queryset(self):
+        return ContentNodeFullTextSearch.objects.select_related("contentnode")
```

Comment:
Since this is a `ValuesViewset`, I don't think having `select_related` is necessary here